.github/workflows/ci.yml

```diff
@@ -37,6 +34,67 @@ jobs:
            </servers>
           </settings>
           EOF
+      - name: Cache Maven packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Clone CCM
+        uses: actions/checkout@v2
+        with:
+          repository: riptano/ccm
+          ref: master
+          fetch-depth: 1
+          path: ccm
+      - name: Install CCM
+        run: |
+          cd ccm
+          sudo ./setup.py install
+      - name: run tests
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          CCM_CLUSTER_START_TIMEOUT_OVERRIDE: 600
+        run: |
+          MAVEN_OPTS="-XX:+TieredCompilation -XX:TieredStopAtLevel=1" # speed up jvm
+          mvn -B verify --file pom.xml  -P it-cassandra-3.11
+  build-cassandra-4_0:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1.4.2
+        with:
+          java-version: 1.8
+      - name: Setup Maven
+        env:
+          ARTIFACTORY_USERNAME: ${{ secrets.ARTIFACTORY_USERNAME }}
+          ARTIFACTORY_PASSWORD: ${{ secrets.ARTIFACTORY_PASSWORD }}
+        run: |
```

Comment:
isn't there a way to extract this artifactory setup logic and share between builds? (it seems that it is the same for this and `dse-6_8`)

Reply:
GH actions do not support YAML anchors :shrug: